c10/core/TensorOptions.h

```diff
@@ -621,9 +621,9 @@ inline DispatchKey computeDispatchKey(TensorOptions options) {
 }
 
 inline DeviceType computeDeviceType(DispatchKey tid) {
-  if (tid == DispatchKey::CPU) {
+  if (tid == DispatchKey::CPU || tid == DispatchKey::AutogradCPU) {
```

Comment:
I'm a little worried about using `AutogradCPU` and `AutogradCUDA` and `AutogradXLA` to detect their respective backends - it adds an asymmetry where backends without backend-specific autograd keys aren't visible to the system in the same way, so if somebody builds logic that depends on detecting backend from *only* the autograd key, it's only going to work for some backends. I think it might be more consistent just to ignore Autograd keys when detecting backend. WDYT?